tokio/src/runtime/task/core.rs

```diff
@@ -107,7 +107,7 @@ impl<T: Future, S: Schedule> Cell<T, S> {
             },
             core: Core {
                 scheduler: Scheduler {
-                    scheduler: UnsafeCell::new(None),
+                    scheduler: UnsafeCell::new(Some(scheduler)),
```

Comment:
Is this ever `None` anymore?

If it is never `None`, you don't need to remove `Option` in this PR but we should clean it up in a follow up.

Reply:
I intentionally did not include that in this PR because the diff is already really long, and it wasn't that simple to remove the `None` case.